code/modules/recycling/conveyor.dm

```diff
@@ -113,67 +157,63 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	icon_state = "[base_icon_state][(machine_stat & BROKEN) ? "-broken" : (operating * verted)]"
 	return ..()
 
+/obj/machinery/conveyor/proc/set_operating(new_value)
+	if(operating == new_value)
+		return
+	operating = new_value
+	update_appearance()
+	update_move_direction()
+	if(!operating) //If we ever turn off, disable moveloops
+		for(var/atom/movable/movable in get_turf(src))
+			stop_conveying(movable)
+
 /obj/machinery/conveyor/proc/update()
+	. = TRUE
 	if(machine_stat & BROKEN || !operable || machine_stat & NOPOWER)
-		operating = FALSE
-		update_appearance()
+		set_operating(FALSE)
 		return FALSE
-	return TRUE
-
-// machine process
-// move items to the target location
-/obj/machinery/conveyor/process()
-	if(machine_stat & (BROKEN | NOPOWER))
+	if(operating) //If we're on, start conveying so moveloops on our tile can be refreshed if they stopped for some reason
+		for(var/atom/movable/movable in get_turf(src))
+			start_conveying(movable)
+
+/obj/machinery/conveyor/proc/conveyable_enter(datum/source, atom/convayable)
+	SIGNAL_HANDLER
+	if(!CAN_CONVEY(src))
+		SSmove_manager.stop_looping(convayable, SSconveyors)
 		return
-
-	//If the conveyor is broken or already moving items
-	if(!operating || conveying)
+	var/datum/move_loop/move/moving_loop = SSmove_manager.processing_on(convayable, SSconveyors)
+	if(moving_loop)
+		moving_loop.direction = movedir
 		return
+	start_conveying(convayable)
 
-	use_power(6)
+/obj/machinery/conveyor/proc/conveyable_exit(datum/source, atom/convayable, direction)
+	SIGNAL_HANDLER
+	var/has_conveyor = neighbors["[direction]"]
+	if(!has_conveyor || !isturf(convayable.loc)) //If you've entered something on us, stop moving
+		SSmove_manager.stop_looping(convayable, SSconveyors)
 
-	//get the first 30 items in contents
-	var/turf/locturf = loc
-	var/list/items = locturf.contents - src
-	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
+/obj/machinery/conveyor/proc/start_conveying(atom/movable/moving)
+	if(!istype(moving) || iseffect(moving) || isdead(moving) || moving == src)
 		return
-	var/list/affecting
-	if(length(items) > MAX_CONVEYOR_ITEMS_MOVE)
-		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1)//Lists start at 1 lol
-	else
-		affecting = items
-	conveying = TRUE
-
-	addtimer(CALLBACK(src, .proc/convey, affecting), 1)//Movement effect
+	moving.AddComponent(/datum/component/convey, movedir, 0.2 SECONDS)
```

Comment:
i wonder if you can make this into just a dedicated movement packet instead of a component

Reply:
It would show as an issue if someone somehow set things up to move/remove things from a conveyor really fast. But that's hard to do, and doesn't seem to be too bad? even in testing.

I'm trying to keep packets as pure as possible, rather then having them manage everything. This makes some things easier (space movement ending lag for instance) and I hope keeps things a bit cleaner

Reply:
On the above note, this is why I haven't converted disposals despite them using sleep loops. They'd need their own unique move type just for them, still working out how I want to handle that